lib/features/registration/pages/credit_card_details_page.dart

```diff
@@ -8,95 +8,141 @@ import 'package:givt_app/features/auth/cubit/auth_cubit.dart';
 import 'package:givt_app/features/family/app/family_pages.dart';
 import 'package:givt_app/features/family/app/injection.dart';
 import 'package:givt_app/features/family/features/home_screen/cubit/navigation_bar_home_cubit.dart';
+import 'package:givt_app/features/family/shared/widgets/errors/retry_error_widget.dart';
+import 'package:givt_app/features/family/shared/widgets/loading/full_screen_loading_widget.dart';
 import 'package:givt_app/features/permit_biometric/models/permit_biometric_request.dart';
 import 'package:givt_app/features/registration/bloc/registration_bloc.dart';
 import 'package:givt_app/features/registration/cubit/stripe_cubit.dart';
 import 'package:givt_app/utils/analytics_helper.dart';
 import 'package:givt_app/utils/stripe_helper.dart';
 import 'package:go_router/go_router.dart';
 
-class CreditCardDetailsPage extends StatelessWidget {
-  const CreditCardDetailsPage({
+class CreditCardDetails extends StatefulWidget {
+  const CreditCardDetails({
+    this.shrink = false,
+    this.navigate = true,
     super.key,
   });
+  final bool shrink;
+  final bool navigate;
+  @override
+  State<CreditCardDetails> createState() => _CreditCardDetailsState();
+
+  static void show(
+    BuildContext context, {
+    bool shrink = false,
+    bool navigate = true,
+  }) {
+    showModalBottomSheet<void>(
+      context: context,
+      isScrollControlled: true,
+      shape: RoundedRectangleBorder(
+        borderRadius: BorderRadius.circular(10),
+      ),
+      backgroundColor: Colors.white,
+      builder: (context) => CreditCardDetails(
+        shrink: shrink,
+        navigate: navigate,
+      ),
+    );
+  }
+}
+
+class _CreditCardDetailsState extends State<CreditCardDetails> {
+  @override
+  void initState() {
+    super.initState();
+    getIt<StripeCubit>().fetchSetupIntent();
+    context.read<RegistrationBloc>().add(const RegistrationInit());
```

Comment:
Why are you adding this init event? When I look at where the code in the RegistrationBloc goes to I don't think it helps you in any way?

Reply:
I am adding it because it clears the Registration state, it was the case that if a user added details successfully then second time they use the sheet it would not clear bc the listener would not get triggered. Though, thinking about it it is unlikely that they will have a successful case and then see this sheet again. But it happened to me while testing